crates/orderbook/src/api.rs

```diff
@@ -198,6 +198,13 @@ impl IntoWarpReply for PriceEstimationError {
                 error("ZeroAmount", "Please use non-zero amount field"),
                 StatusCode::BAD_REQUEST,
             ),
+            Self::UnsupportedOrderType => with_status(
+                error(
+                    "UnsupportedOrderType",
+                    "Can not estimate price for this order type",
+                ),
+                StatusCode::BAD_REQUEST,
+            ),
```

Comment:
Imo: In our top level api we always want to support all order types. That some price estimators cannot support buy orders is an implementation detail. So I would treat this like `Other`.
Likewise in competition we should give Other priority over unsupported.

Reply:
That's a good point. An internal server error seems more fitting than saying the order type is not supported at all.

Reply:
Good point. I also naively assumed `BAD_REQUEST` here. Maybe we can add a comment about this in code as it wasn't immediately obvious to me.